### ISSUE TITLE
[Question] Typings for AwsAppsync, expose Swift functions to NativeScript.

### ISSUE THREAD

**Issue by aaayushsingh** [NativeScript/ios-runtime#984]
I'm trying to generate typings for` aws-appsync-ios-sdk`. I added `@objc    ...    : Int {` around enums but most of the functions are still missing in the typings file. Is there something else I need to do as well?

**Comment by aaayushsingh** [comment]
[This](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/blob/master/AWSAppSyncClient/AWSAppSyncClient.swift) is the file I'm trying to generate typings for. I only get [these functions in output](https://gist.github.com/aaayushsingh/04f2252fc7019e8df7da55f54f975194)

By adding `@objc ... : Int { ` around enums I just got the enum at the end of the file extra.

Am I missing something? How do I generate these typings correctly, and how do I access these `Swift` functions from `JS`

**Comment by mbektchiev** [comment]
Hi @aaayushsingh, I'm sorry for the delayed response. Here's what I found out:

* According to the [Swift Migration Guide](https://developer.apple.com/documentation/swift/migrating_your_objective-c_code_to_swift) a class needs to inherit from `NSObject` or some other Objective-C class:
> To make your Swift class accessible and usable back in Objective-C, make it a descendant of an Objective-C class.

* Inheriting from `NSObject` however seems to be insufficient for making a class accessible from ObjC, the other prerequisite that I discovered is that it has to have `public` or `open` access. The classes in this file are `internal` since this is the default when no access specifier is present. This requirement seems quite obvious if you examine the [Access Control in Swift article](https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html):
> *Internal access* enables entities to be used within any source file from their defining module, but not in any source file outside of that module. You typically use internal access when defining an app’s or a framework’s internal structure.

So if you change the access level of all classes and methods that you'd like to use to `public` and make the classes inherit from `NSObject`, you'll be able to generate typings and metadata, and thus make them callable from NativeScript
